bigquery/google/cloud/bigquery/job.py

```diff
@@ -1949,6 +1953,29 @@ def result(self, timeout=None, retry=DEFAULT_RETRY):
         return self._client.list_rows(dest_table, selected_fields=schema,
                                       retry=retry)
 
+    def to_dataframe(self):
+        """Create a pandas DataFrame from the query results.
+
+        Returns:
+            A :class:`~pandas.DataFrame` populated with row data and column
+            headers from the query results. The column headers are derived
+            from the destination table's schema.
+
+        Raises:
+            ValueError: If the `pandas` library cannot be imported.
+
+        """
+        if pandas is None:
+            raise ValueError('The pandas library is not installed, please '
+                             'install pandas to use the to_dataframe() '
+                             'function.')
+
+        query_results = self.result()
+        column_headers = [field.name for field in query_results.schema]
+        rows = [row.values() for row in query_results]
+
+        return pandas.DataFrame(rows, columns=column_headers)
```

Comment:
Have you tried this on a query that includes some nulls in results? I'm curious if we have to do anything special to tell Pandas when there might be nulls. Also curious about what types Pandas ends up using. It'd be good to see a unit test with some rows that are timestamp values.

Reply:
I have a bunch of type testing I did in a notebook that I can move into the unit tests. Any other types we should cover in unit tests besides timestamps?

Reply:
Based on https://stackoverflow.com/a/21767006/101923

Pandas supports:
datetime64, timedelta64, int64, float64, bool and object

So, we should try to cover at least these.

I don't believe BigQuery has a native timedelta type, since a [timestamp diff comes back as an int64](https://cloud.google.com/bigquery/docs/reference/standard-sql/functions-and-operators#timestamp_diff).

Reply:
I added a test to check handling for null and most of the BigQuery data types. I used the [standard SQL data types](https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types) for reference. Should I add tests for array and/or struct types?

Reply:
Pandas doesn't really have great support for those data types AFAIK, but I suppose it would be good to verify that Pandas doesn't mess them up (falling back to the "object" data type in Pandas)